Some additions cleaned up

aaronliu0130 opened the pull request:
Basically #1, but cleaned up.

aaronliu0130 committed:
Add some new codes

Cleanup

aaronliu0130 committed:
Add two programs

aaronliu0130 committed:
Add build files

Make project easier to build.

aaronliu0130 committed:
Fix bugs

Fix some bugs. Thank you @travis-ci !

aaronliu0130 committed:
Update .travis.yml

Add support for linux

jerronl commented:
Great effort. Several thoughts:

- You made so many changes! A good effort. However you put too many changes in one request. It seems they are for different independent problems. If so, can you divide them into separate requests? It then would allow me to approve some of them and reject the others.
- Some of the files seems generated by the building tools, such as the .mk files. Put them into .gitignore files. What would be pushed into the repository should be the minimum subset allowing others to reproduce.  BTW, minimize the .gitignore files, too. If one in the parent folder would do, remove the ones in the sub folders.
- In the beginning of the cpp file for each problem, can we keep the description lines like following:
```
https://www.lintcode.com/problem/binary-tree-level-order-traversal/description
Given a binary tree, return the level order traversal of its nodes' values. 
```
It would be convenient to the viewers.
 
- What is travis?

jerronl commented on lintcode/utils/utils.h:
```diff
@@ -7,12 +7,16 @@
 #ifndef UTIL_H_
 #define UTIL_H_
-
+#ifdef __linux__
```
Why do you want to define INT_MIN? It is defined in limits.h. Does it cause issues? If so, the right solution is probably using the exact constant explicitly to avoid confusion:
https://stackoverflow.com/questions/3233054/error-int32-max-was-not-declared-in-this-scope

Putting the redefinition like this would introduce a potential bug: the order you include heading file would cause different behavior. Imagine you include utils.h first before limits.h vs the other way.

aaronliu0130 commented:
> * Some of the files seems generated by the building tools, such as the .mk files. Put them into .gitignore files. What would be pushed into the repository should be the minimum subset allowing others to reproduce.

Well, I couldn't figure out the commands that the makefiles do, so I had included the makefiles to build the repository.
> * What is travis?

<details><summary>Short answer: Travis can help me build and test my repository.</summary></details>
<details>
  <summary>Long answer</summary>

  Travis CI is a hosted continuous integration service used to build and test software projects hosted at GitHub. Travis CI provides various paid plans for private projects and a free plan for open source projects. Travis CI enables your team to test and ship your apps with confidence. It’s built for everyone and for projects and teams of all sizes, supporting over 20 different languages out of the box, including Javascript and Node.js, Ruby, PHP, Python, Mac/iOS, as well as Docker, while giving you full control over the build environment to customize it to your own needs.

</details>
I can do the other two and minimize the `.gitignore` files though. Working on it...

aaronliu0130 committed:
Simplify .gitignore

Down to the bone

aaronliu0130 commented:
Wait. I still don't know how to create a new branch from history, so I can't address the first bullet point yet.

aaronliu0130 committed:
Add descriptive titles

Commit:
Delete Make

Make travis respond, add build troubleshooting, simplify gitignore, delete make files

aaronliu0130 committed:
Cleanup

Better info files
